oauth2client/client.py

```diff
@@ -272,7 +272,7 @@ def new_from_json(cls, s):
       An instance of the subclass of Credentials that was serialized with
       to_json().
     """
-    if six.PY3 and isinstance(s, bytes):
```

Comment:
I'm not sure I understand this change - why drop the six.PY3 conjunct? Under what circumstances will s be of type bytes? Should the method's doc string be updated to explicitly allow bytes to be passed? As a lifelong opponent of ad hoc polymorphism, help me understand why explicit support for s being either a string or bytes is a good thing?